lrc_file/LrcFile.py

```diff
@@ -218,8 +261,10 @@ def _evaluation_data(self, result_type: type):
 
                 evaluated_metric = comparison["metric_name"]
                 evaluated_metric_name = evaluated_metric[4:]
-
-                evaluation_data[evaluated_metric] = getattr(comparison["current_result"], evaluated_metric_name).average
+                if result_type is TcRunMeasurementResults:
```

Comment:
I have some concerns about current implementation of the TcRunMeasurement and I think we should talk about that first before we start adopting changes in dependant projects.

I understand this is probably the only solution with the current implementation but this brought my attention and I took a look at the rest of the code and I think we're doing this very complicated way.

Adopting changes with the current code may bring in more confusion and complexity in other projects.

I'll provide at least the high level on what I'm concerned about.

The `PerfInterval` that we use for performance measurements are designed for time based samples of some metric (packets sent) for specific duration. So, we either take a sample of a metric every second (iperf-throughput) or we take metric value of the whole duration of a performance measurement (iperf-cpu).

But with TcRunMeasurement
* we do not know what is the duration before hand and it is in a sense irrelevant
* the metric we want to record and evaluate is time-taken-to-add-num-of-rules and it is simply not sampled the same way as for example the iperf-throughput (bits/sec)

I think we should define a new type of result that would take this into consideration. We could still reuse most of the abstractions, but we'll need rethink the atomic objects first.

Reply:
I thought of `TimedResult(TimedStatMixin)` that would be similar to `PerfResult`.

TimedResult would be representing anything that matches pattern of Linux command: `time some_command`.

Reply:
@jtluka I sorta see what your saying and agree with you, I kinda was thinking the same thing after finishing the lnst side of things. Lets discuss this tomorrow.

Reply:
We already discussed this with LNST team and changes were made to LNST in https://github.com/LNST-project/lnst/pull/293. Resolving.